packages/lit-html/src/lit-html.ts

```diff
@@ -750,11 +770,18 @@ class TemplateInstance {
     const {
       _$element: {content},
       _parts: parts,
+      _hostWrapper,
     } = this._$template;
     const fragment = d.importNode(content, true);
     walker.currentNode = fragment;
 
-    let node = walker.nextNode();
+    if (DEV_MODE) {
+      if (_hostWrapper && !(options?.host instanceof HTMLElement)) {
+        throw new Error('When using <host> in template, RenderOptions must include a `host` property set to the host element');
+      }
```

Comment:
Should we also check that <host> is present at most once, and as the sole root nonempty Node in the template?